include/pybind11/stl_binders.h

```diff
@@ -0,0 +1,226 @@
+
+	}
+	template<typename U = T, typename std::enable_if< !has_insertion_operator<U>() >::type * = nullptr>
+	void maybe_has_insertion_operator(Class_ &, char const *) {}
+	#endif
+
+
+
+public:
+	vector_binder(pybind11::module &m, char const *name) {
+		Class_ cl(m, name);
+
+		cl.def(pybind11::init<>());
+
+		//maybe_constructible(cl);
+		maybe_default_constructible(cl);
+		maybe_copy_constructible(cl);
+
+		// Element access
+		cl.def("front", [](Vector &v) {
+				if( v.size() ) return v.front();
+				else throw pybind11::index_error();
+			}, "access the first element");
+		cl.def("back", [](Vector &v) {
+				if( v.size() ) return v.back();
+				else throw pybind11::index_error();
+			}, "access the last element ");
+		// Not needed, the operator[] is already providing bounds checking cl.def("at", (T& (Vector::*)(SizeType i)) &Vector::at, "access specified element with bounds checking");
+
+
+		// Capacity
+		cl.def("empty",         &Vector::empty,         "checks whether the container is empty");
+		cl.def("size",          &Vector::size,          "returns the number of elements");
+		cl.def("max_size",      &Vector::max_size,      "returns the maximum possible number of elements");
+		cl.def("reserve",       &Vector::reserve,       "reserves storage");
+		cl.def("capacity",      &Vector::capacity,      "returns the number of elements that can be held in currently allocated storage");
+		cl.def("shrink_to_fit", &Vector::shrink_to_fit, "reduces memory usage by freeing unused memory");
+
+		// Modifiers, C++ style
+		cl.def("clear",                                      &Vector::clear, "clears the contents");
+		cl.def("push_back", (void (Vector::*)(const T&)) &Vector::push_back, "adds an element to the end");
+		cl.def("pop_back",                                &Vector::pop_back, "removes the last element");
+		cl.def("swap",                                        &Vector::swap, "swaps the contents");
+
+
+		// Modifiers, Python style
+		cl.def("append", (void (Vector::*)(const T&)) &Vector::push_back, "adds an element to the end");
+		cl.def("insert", [](Vector &v, SizeType i, const T&t) {v.insert(v.begin()+i, t);}, "insert an item at a given position");
+		cl.def("pop", [](Vector &v) {
+				if( v.size() ) {
+					T t = v.back();
+					v.pop_back();
+					return t;
+				}
+				else throw pybind11::index_error();
+			}, "insert an item at a given position");
+
+
+		cl.def("erase", [](Vector &v, SizeType i) {
+				if( i >= v.size() ) throw pybind11::index_error();
+				v.erase( v.begin() + i );
+			}, "erases element at index");
+
+
+		// Python friendly bindings
+		#ifdef PYTHON_ABI_VERSION // Python 3+
+			cl.def("__bool__",    [](Vector &v) -> bool { return v.size() != 0; }); // checks whether the container has any elements in it
+		#else
+			cl.def("__nonzero__", [](Vector &v) -> bool { return v.size() != 0; }); // checks whether the container has any elements in it
+		#endif
+
+		cl.def("__getitem__", [](Vector const &v, SizeType i) {
```

Comment:
Slice based __getitem__ and __setitem__ would be important to have. See ``example/example6.cpp`` for reference.